artemis-jdbc-store/src/test/java/org/apache/activemq/artemis/jdbc/store/sql/PropertySQLProviderTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.jdbc.store.sql;
+
+import java.sql.DriverManager;
+
+import org.apache.activemq.artemis.tests.util.ArtemisTestCase;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+import static org.apache.activemq.artemis.jdbc.store.sql.PropertySQLProvider.Factory.SQLDialect.DERBY;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class PropertySQLProviderTest extends ArtemisTestCase {
+
+   @Test
+   public void testGetProperty() {
+      for (SQLProvider.DatabaseStoreType storeType : SQLProvider.DatabaseStoreType.values()) {
+         PropertySQLProvider factory = (PropertySQLProvider) new PropertySQLProvider.Factory(DERBY).create("who-cares", storeType);
+         factory.sql("create-file-table");
+      }
+   }
+
+   @Test
+   public void testGetMissingProperty() {
+      for (SQLProvider.DatabaseStoreType storeType : SQLProvider.DatabaseStoreType.values()) {
+         try {
+            PropertySQLProvider factory = (PropertySQLProvider) new PropertySQLProvider.Factory(DERBY).create("who-cares", storeType);
+            factory.sql("foo");
+            fail();
+         } catch (IllegalStateException e) {
+            // expected
+         }
+      }
+   }
+
+   @AfterEach
+   public void shutdownDerby() {
+      try {
+         DriverManager.getConnection("jdbc:derby:;shutdown=true");
+      } catch (Exception ignored) {
+      }
```

Comment:
Probably nicer just to call DBSupportUtil._shutdownDerby_, which also e.g verifies the shutdown exception is as expected.

Reply:
Hmmm...For some reason this change succeeding locally but blowing up on the PR build.

Reply:
Interesting. Doesnt the failure suggest the driver was never registered in the test to begin with (I just assumed it was, given you added the shutdown step)? If so maybe it passing for you but failing here means its dependent on the behaviour of other tests classes, or even the order of tests being run? When you say 'locally' do you mean in-IDE, or on CLI (where order should be the same, unless particular profiles influence the tests run)? I do see the earlier tests in the PR build all look to have their own bespoke Derby shutdown handling (some similarly swallowing all exceptions and thus not knowing if they got the shutdown-succeeded one)

Presumably also means your shutdown attempt was also failing originally, just being swallowed? Which was partly what made me think of using the utility method to ensure it wasnt failing. And also as I thought it had more options to stop the shutdown deregistering the driver, in case other tests use it later.

Depending on the situation it might be a case of having to swallow the failure though, to cover off the alternative of never shutting it down if it is later actually registered/used due to code changes?

Reply:
> Doesnt the failure suggest the driver was never registered in the test to begin with (I just assumed it was, given you added the shutdown step)?

Yes. I originally copied this test from `org.apache.activemq.artemis.jdbc.store.journal.JDBCJournalLoaderCallbackTest` which had the `shutdownDerby` method. Apparently this isn't needed at all and in that test it swallows a similar exception every time. I changed my tests to use a `null` SQL dialect to make it more clear that Derby isn't required.